File: archive-application/src/main/java/site/archive/service/user/UserProfileImageService.java
package site.archive.service.user;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;
import site.archive.domain.user.UserRepository;

@Service
@Transactional(readOnly = true)
@RequiredArgsConstructor
public class UserProfileImageService {

    private final UserRepository userRepository;

    @Transactional
    public void updateUserProfileImage(Long userId, String profileImageUri) {
        userRepository.updateUserProfileImage(userId, profileImageUri);
    }

}

File: archive-application/src/main/kotlin/site/archive/service/user/UserProfileImageService.kt
package site.archive.service.user

import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import site.archive.domain.user.UserRepository

@Service
@Transactional(readOnly = true)
class UserProfileImageService(val userRepository: UserRepository) {

    @Transactional
    fun updateUserProfileImage(userId: Long, profileImageUri: String) {
        userRepository.updateUserProfileImage(userId, profileImageUri)
    }

}